src/generators/legacy-json/utils/parseList.mjs

```diff
@@ -0,0 +1,114 @@
+import {
+  DEFAULT_EXPRESSION,
+  LEADING_HYPHEN,
+  NAME_EXPRESSION,
+  RETURN_EXPRESSION,
+  TYPE_EXPRESSION,
+} from '../constants.mjs';
+import parseSignature from './parseSignature.mjs';
+import { transformNodesToString } from '../../../utils/unist.mjs';
+
+/**
+ * Transforms type references in a string by replacing template syntax with curly braces and cleaning up.
+ * @param {String} string
+ * @returns {String}
+ */
+function transformTypeReferences(string) {
+  return string.replace(/`<([^>]+)>`/g, '{$1}').replaceAll('} | {', '|');
+}
+
+/**
+ * Extracts a matching pattern from a text and assigns it to the current object.
+ * @param {string} text
+ * @param {RegExp} pattern
+ * @param {string} key
+ * @param {Object} current
+ * @returns {string}
+ */
+const extractPattern = (text, pattern, key, current) => {
+  const match = text.match(pattern)?.[1]?.trim().replace(/\.$/, '');
+  if (match) {
```

Comment:
I believe this can be simplified

Reply:
I don't think it can? What do you have in mind?

Reply:
I rewrote it, but it's not simplified, LMK what you think

Reply:
lgtm